nio/client/async_client.py

```diff
@@ -2928,3 +2933,60 @@ async def upload_filter(
         )
 
         return await self._send(UploadFilterResponse, method, path, data)
+
+    @logged_in
+    async def set_pushrule(
+        self,
+        scope: str,
+        kind: PushRuleKind,
+        rule_id: str,
+        before: Optional[str] = None,
+        after: Optional[str] = None,
+        actions: Sequence[PushAction] = (),
+        conditions: Optional[Sequence[PushCondition]] = None,
+        pattern: Optional[str] = None,
+    ) -> Union[SetPushRuleResponse, SetPushRuleError]:
+        """Create or modify an existing push rule.
+
+        Returns either a `SetPushRuleResponse` if the request was
+        successful or a `SetPushRuleError` if there was an error
+        with the request.
+
+        Args:
+            scope (str): The scope of this rule, e.g. ``"global"``.
```

Comment:
Are there more scopes than global? Can they be arbitrarily defined? We should mention this here.